week4/mission2/src/apis/axios.ts

```
@@ -1,11 +1,96 @@
+// 응답 인터셉터 - 401 발생 시 Refresh Token으로 토큰 갱신 시도
+axiosInstance.interceptors.response.use(
+    (response) => response,
+    async (error) => {
+        const originalRequest: CustomInternalAxiosRequestConfig = error.config;
+
+        // 401 + 재시도한 적 없는 요청만 처리
+        if (error.response && error.response.status === 401 && !originalRequest._retry) {
+            // Refresh API 자체가 401이면 강제 로그아웃
+            if (originalRequest.url === "/v1/auth/refresh") {
+                const { removeItem: removeAccessToken } = useLocalStorage(LOCAL_STORAGE_KEY.accessToken);
+                const { removeItem: removeRefreshToken } = useLocalStorage(LOCAL_STORAGE_KEY.refreshToken);
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)